leshan-client-demo/src/main/java/org/eclipse/leshan/client/demo/LeshanClientDemo.java

```diff
@@ -312,20 +321,100 @@ public static void main(final String[] args) {
                 return;
             }
         }
+
         // Set boolean controlling OSCORE usage
-        boolean useOSCore = false;
+        OSCoreSettings oscoreSettings = null;
         if (cl.hasOption("oscore")) {
             System.out.println("Using OSCORE");
-            useOSCore = true;
 
             HashMapCtxDB db = OscoreHandler.getContextDB();
             OSCoreCoapStackFactory.useAsDefault(db);
 
+            // Parse OSCORE related command line parameters
+
+            String mastersecretStr = cl.getOptionValue("mastersecret");
+            String mastersaltStr = cl.getOptionValue("mastersalt");
+            String idcontextStr = cl.getOptionValue("idcontext");
+            String senderidStr = cl.getOptionValue("senderid");
+            String recipientidStr = cl.getOptionValue("recipientid");
+            String aeadStr = cl.getOptionValue("aead");
+            String hkdfStr = cl.getOptionValue("hkdf");
+
+            if (mastersecretStr == null) {
+                System.err.println("The OSCORE master secret must be indicated");
+                formatter.printHelp(USAGE, options);
+                return;
+            }
+
+            // Set salt to empty string if not indicated
+            if (mastersaltStr == null) {
+                mastersaltStr = "";
+            }
+
+            if (idcontextStr != null) {
+                System.err.println("The OSCORE ID Context parameter is not yet supported");
+                formatter.printHelp(USAGE, options);
+                return;
+            }
+
+            if (senderidStr == null) {
+                System.err.println("The OSCORE Sender ID must be indicated");
+                formatter.printHelp(USAGE, options);
+                return;
+            }
+
+            // Check that the Sender ID matches the client name
+            // http://www.openmobilealliance.org/release/LightweightM2M/V1_1-20180710-A/HTML-Version/OMA-TS-LightweightM2M_Transport-V1_1-20180710-A.html#5-5-5-0-555-Endpoint-Client-Name
+            if (!Arrays.equals(endpoint.getBytes(), Hex.decodeHex(senderidStr.toCharArray()))) {
+                System.err.println(
+                        "The endpoint client name (" + endpoint + ") must be equal to the OSCORE Sender ID (use '"
+                                + Hex.encodeHexString(endpoint.getBytes()).toUpperCase() + "')");
+                formatter.printHelp(USAGE, options);
+                return;
+            }
+
+            if (recipientidStr == null) {
+                System.err.println("The OSCORE Recipient ID must be indicated");
+                formatter.printHelp(USAGE, options);
+                return;
+            }
+
+            // Parse AEAD Algorithm (set default if not indicated)
+            if (aeadStr == null) {
+                aeadStr = "AES_CCM_16_64_128";
+            }
+
+            int aeadInt;
+            if (aeadStr.equals("AES_CCM_16_64_128") || aeadStr.equals("10")) {
```

Comment:
if we support only 1 AEAD algorithmm we can remove this option ? and just document that only this algorithm is used.

Reply:
I have now updated the parsing of the AEAD algorithm to parse it into a COSE AlgorithmID and not compare it to fixed values. My goal is to support multiple algorithms. Done in commit 76fa1268a8f8262eb39008693f37a75069173ba2